sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/in_memory_checkpoint_store.py

```diff
@@ -0,0 +1,142 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+from typing import List, Dict, Any, Iterable, Optional, Union
+import time
+import uuid
+import logging
+
+from .checkpoint_store import CheckpointStore
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class _TrieNode(object):
+    def __init__(self, value, is_leaf):
+        self.value = value
+        self.is_leaf = is_leaf
+        if is_leaf:
+            self.children = None
+        else:
+            self.children = {}
+
+    def __contains__(self, item):
+        if self.is_leaf:
+            raise ValueError("Please don't call __contains__ on a leaf TrieNode")
+        return item in self.children
+
+    def get(self, key):
+        if self.is_leaf:
+            raise ValueError("Please don't get a child from a leaf TrieNode")
+        return self.children.get(key)
+
+    def put(self, key, value):
+        if self.is_leaf:
+            raise ValueError("Please don't put a value to a leaf TrieNode")
+        self.children[key] = value
+
+
+def _lookup_trie(root, path, path_index):
+    if path_index == len(path) - 1:
+        return root.get(path[path_index])
+    if path[path_index] in root:
+        return _lookup_trie(root.children[path[path_index]], path, path_index+1)
+    return None
+
+
+def _list_leaves_trie(root, result):
+    if root.is_leaf:
+        result.append(root.value)
+        return
+    for child in root.children.values():
+        _list_leaves_trie(child, result)
+
+
+def _set_ele_trie(root, ele, path_names, path_index):
+    if path_index == len(path_names) - 1:
+        root.children[ele[path_names[path_index]]] = _TrieNode(ele, True)
+        return
+    next_node = root.get(ele[path_names[path_index]])
+    if not next_node:
+        next_node = _TrieNode(ele[path_names[path_index]], False)
+        root.put(ele[path_names[path_index]], next_node)
+    _set_ele_trie(next_node, ele, path_names, path_index + 1)
```

Comment:
```key_name = path_names[path_index]```

Nit: `def _lookup_trie(root, path, path_index):` has param `path`, while here it is named `path_names`